From: Michael Catanzaro <mcatanzaro@redhat.com>
Date: Thu, 8 May 2025 09:27:01 -0500
Subject: auth-digest: fix crash in soup_auth_digest_get_protection_space()

We need to validate the Domain parameter in the WWW-Authenticate header.

Unfortunately this crash only occurs when listening on default ports 80
and 443, so there's no good way to test for this. The test would require
running as root.

Fixes #440
---
 libsoup/soup-auth-digest.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libsoup/soup-auth-digest.c b/libsoup/soup-auth-digest.c
index fbd216a..625bf78 100644
--- a/libsoup/soup-auth-digest.c
+++ b/libsoup/soup-auth-digest.c
@@ -229,7 +229,7 @@ soup_auth_digest_get_protection_space (SoupAuth *auth, SoupURI *source_uri)
 			uri = soup_uri_new (d);
 			if (uri && uri->scheme == source_uri->scheme &&
 			    uri->port == source_uri->port &&
-			    !strcmp (uri->host, source_uri->host))
+			    !g_strcmp0 (uri->host, source_uri->host))
 				dir = g_strdup (uri->path);
 			else
 				dir = NULL;
